third_party/move/move-vm/runtime/src/interpreter.rs

```diff
@@ -1159,7 +1149,7 @@ impl Frame {
             Bytecode::Ret => {
                 for (idx, ty) in local_tys.iter().enumerate() {
                     if !locals.is_invalid(idx)? {
-                        check_ability(resolver.loader().abilities(ty)?.has_drop())?;
+                        check_ability(ty.abilities()?.has_drop())?;
```

Comment:
For my eduction: these are all paranoid checks, right? Definitely good choice to make the abilities part of the type, specifically if we start to intern the types (which we should, but not as part of this PR).

Reply:
+1